AK/BitStream.h

```diff
@@ -121,6 +122,85 @@ class InputBitStream final : public InputStream {
     InputStream& m_stream;
 };
 
+class OutputBitStream final : public OutputStream {
+public:
+    explicit OutputBitStream(OutputStream& stream)
+        : m_stream(stream)
+    {
+    }
+
+    // WARNING: write aligns to the next byte boundary before writing, if unaligned writes are needed this should be rewritten
+    size_t write(ReadonlyBytes bytes) override
+    {
+        if (has_any_error())
+            return 0;
+        align_to_byte_boundary();
+        if (has_fatal_error()) // if align_to_byte_boundary failed
+            return 0;
+        return m_stream.write(bytes);
+    }
+
+    bool write_or_error(ReadonlyBytes bytes) override
+    {
+        if (write(bytes) < bytes.size()) {
+            set_fatal_error();
+            return false;
+        }
+        return true;
+    }
+
+    void write_bits(u32 bits, size_t count)
+    {
+        VERIFY(count <= 32);
+        size_t n_written = 0;
+        while (n_written < count) {
+            if (m_stream.has_any_error()) {
+                set_fatal_error();
+                return;
+            }
+
+            if (m_next_byte.has_value()) {
+                m_next_byte.value() |= ((bits >> n_written) & 1) << m_bit_offset;
```

Comment:
Optimizing this routine to do more than a bit at a time would probably give a good speed improvement.